src/form.js

```diff
@@ -308,6 +309,11 @@ define([
                         meta = null;
                     }
                     this.knownInstances[attrs.id] = attrs.src;
+                } else if (meta.attributes.id !== attrs.id) {
+                    // support renaming when there is no reference
```

Comment:
Works because the reference is dropped then renamed then readded to the mug. Used in model iteration